lib/librato/collector/counter_cache.rb

```diff
@@ -74,22 +75,28 @@ def flush_to(queue, opts={})
       #   increment :foo, :source => user.id
       #
       def increment(counter, options={})
+        metric = counter.to_s
         if options.is_a?(Fixnum)
           # suppport legacy style
           options = {by: options}
         end
         by = options[:by] || 1
-        if options[:source]
-          metric = "#{counter}#{SEPARATOR}#{options[:source]}"
-        else
-          metric = counter.to_s
+        if options[:tags] && options[:tags].respond_to?(:each)
+          options[:tags].sort.each do |k, v|
+            k = k.is_a?(String) ? k.downcase.delete(' ') : k
+            v = v.is_a?(String) ? v.downcase.delete(' ') : v
+            metric = "#{metric}#{SEPARATOR}#{k}#{SEPARATOR}#{v}"
```

Comment:
re your comment on lib-metrics seems like we do this lots of places and it would be worth rolling into a single utility method somewhere. we should also make sure this is fast because it will get called a ton of times.